### ISSUE TITLE
Accept `process.stdin` as image `--input`

### ISSUE THREAD

**Issue by xerc** [axe312ger/sqip#334]
`sqip -i test.jpg` => `curl -q https://[..]/test.jpg | sqip`

https://github.com/axe312ger/sqip/blob/91f9928c0a06cd3d7fc6a0284bb965a9098f8bfa/packages/sqip-cli/src/sqip-cli.ts#L36-L39
https://github.com/axe312ger/sqip/blob/91f9928c0a06cd3d7fc6a0284bb965a9098f8bfa/packages/sqip-cli/src/sqip-cli.ts#L167-L170

**Comment by axe312ger** [comment]
Good idea!

Would you be so kind to open a PR? Not sure yet when I find time for it.


Thanks!

**Comment by xerc** [comment]
@axe312ger would like to but i get a lot of `DeprecationWarning`s and ERR somewhere with `node-gyp` while `npm install`

**Comment by xerc** [comment]
ref. https://github.com/yarnpkg/yarn/issues/9005 & https://github.com/yarnpkg/yarn/issues/9013

**Comment by axe312ger** [comment]
@xerc the errors you refer to are for node v21.

The project installs fine for me on OSX (on my M1) and Linux (in CI) for node v18 and v20.

Maybe try again with node 18 or 20 :)

**Comment by axe312ger** [comment]
@xerc i got rid of plenty of very old dependencies, can you check if you can clone and build the project now properly on your machine? #342

**Comment by xerc** [comment]
look good besides these two missing packages..
```diff
diff --git package.json
   "devDependencies": {
+    "@types/debug": "4.1.12",
+    "@types/fs-extra": "11.0.4",
```

**Comment by axe312ger** [comment]
@xerc awesome. I'll add them to master right away!

**Comment by axe312ger** [comment]
Okay, thats done as well.

About the process.stdin -> I wanna do this soon, probably also redo some of the logic that reads files and change how meta data is extracted and passed through the plugins.

**Comment by xerc** [comment]
good idea ; incl. the checks for `output` and `print` // if you want to pre-tests the "hacky" way..
`[166]` `args['input'] = Buffer.concat(await process.stdin.toArray()).toString('utf-8');`
